### PR TITLE
Prepend VPC name to VPC network tier name

### PR DESCRIPTION
**Pull request by abh1sar**
### Description

Network tiers which belong to a VPC have their own individual names, in the networks tab they are listed with all other networks, in order to make it easier to identify those networks and correlate them with the VPCs. their names should be prepended with the name of the VPC they belong to.

This PR prepends VPC name to the network tier name using a delimiter.
This feature can be enabled using a global setting which is disabled by default.
The delimiter can also be changed using a global setting. " " by default.

Doc PR : https://github.com/apache/cloudstack-documentation/pull/445

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

Screenshot 2024-10-08 at 11 22 16 AM
Screenshot 2024-10-08 at 11 23 36 AM

### CODE REVIEW

File: engine/components-api/src/main/java/com/cloud/network/vpc/VpcManager.java [hunk 1]
```diff
@@ -37,8 +37,26 @@
 import com.cloud.network.addr.PublicIp;
 import com.cloud.offering.NetworkOffering;
 import com.cloud.user.Account;
+import org.apache.cloudstack.framework.config.ConfigKey;
 
 public interface VpcManager {
+    ConfigKey<Boolean> VpcTierNamePrepend = new ConfigKey<>(Boolean.class,
+            "vpc.tier.name.prepend",
+            "Advanced",
```
Comment (sureshanaparti) [inline engine/components-api/src/main/java/com/cloud/network/vpc/VpcManager.java:45]:
```suggestion
            ConfigKey.CATEGORY_NETWORK,
```

File: engine/components-api/src/main/java/com/cloud/network/vpc/VpcManager.java [hunk 2]
```diff
@@ -37,8 +37,26 @@
...
+            "false",
+            "Whether to prepend the VPC name to the VPC tier network name",
+            true,
+            ConfigKey.Scope.Global,
+            null);
+    ConfigKey<String> VpcTierNamePrependDelimiter = new ConfigKey<>(String.class,
+            "vpc.tier.name.prepend.delimiter",
+            "Advanced",
```
Comment (sureshanaparti) [inline engine/components-api/src/main/java/com/cloud/network/vpc/VpcManager.java:53]:
```suggestion
            ConfigKey.CATEGORY_NETWORK,
```

File: server/src/main/java/com/cloud/network/NetworkServiceImpl.java [hunk 3]
```diff
@@ -1788,6 +1788,15 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
 
         checkNetworkDns(ipv6, ntwkOff, vpcId, ip4Dns1, ip4Dns2, ip6Dns1, ip6Dns2);
 
+        if (vpcId != null && VpcManager.VpcTierNamePrepend.value()) {
+            final String connector = VpcManager.VpcTierNamePrependDelimiter.value();
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/network/NetworkServiceImpl.java:1792]:
```suggestion
            final String delimiter = VpcManager.VpcTierNamePrependDelimiter.value();
```

**Review by sureshanaparti — APPROVED** [review]
clgtm

File: server/src/main/java/com/cloud/network/NetworkServiceImpl.java [hunk 4]
```diff
@@ -1788,6 +1788,14 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
 
         checkNetworkDns(ipv6, ntwkOff, vpcId, ip4Dns1, ip4Dns2, ip6Dns1, ip6Dns2);
 
+        if (vpcId != null && VpcManager.VpcTierNamePrepend.value()) {
+            final String delimiter = VpcManager.VpcTierNamePrependDelimiter.value();
+            Vpc vpc = _vpcDao.findById(vpcId);
+            if (vpc != null) {
+                name = vpc.getName() + delimiter + name;
+            }
+        }
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/NetworkServiceImpl.java:1797]: can this be a method called at the moment `name` is being set?

**Review by DaanHoogland — APPROVED** [review]

File: server/src/main/java/com/cloud/network/NetworkServiceImpl.java (hunk 4 above)
Comment (abh1sar) [reply server/src/main/java/com/cloud/network/NetworkServiceImpl.java]: Done. please check.

**Comment by abh1sar** [comment]
@blueorangutan package

**Comment by abh1sar** [comment]
@blueorangutan test

**Review by alexandremattioli — APPROVED** [review]

**Comment by abh1sar** [comment]
> @abh1sar are there any documentation changes?

@alexandremattioli yes. https://github.com/apache/cloudstack-documentation/pull/445

**Review by kiranchavala — APPROVED** [review]
LGTM

| Test Case Execution | Result |
|---|---|
| Verify the functionality of  global setting   (vpc.tier.name.prepend.delimiter) | Pass |
| Test the global settings at account and   project level  | Pass |
| Verify the functionality of  global setting (vpc.tier.name.prepend) | Pass |

**Pull request merged**